File: server-java-play-framework/app/controllers/FlagApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.CountResponse;
import apimodels.FlagResponse;
import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class FlagApiController extends Controller {
    private final FlagApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private FlagApiController(Config configuration, FlagApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result createFlag(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueflagableType = request.getQueryString("flagableType");
        String flagableType;
        if (valueflagableType != null) {
            flagableType = valueflagableType;
        } else {
            throw new IllegalArgumentException("'flagableType' parameter is required");
        }
        String valueflagableId = request.getQueryString("flagableId");
        Long flagableId;
        if (valueflagableId != null) {
            flagableId = Long.parseLong(valueflagableId);
        } else {
            throw new IllegalArgumentException("'flagableId' parameter is required");
        }
        String valueflagDescription = request.getQueryString("flagDescription");
        String flagDescription;
        if (valueflagDescription != null) {
            flagDescription = valueflagDescription;
        } else {
            flagDescription = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.createFlagHttp(request, version, flagableType, flagableId, deviceId, accountId, flagDescription, latitude, longitude);
    }

    @ApiAction
    public Result deleteFlag(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueitemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType");
        String itemBeingFlaggedType;
        if (valueitemBeingFlaggedType != null) {
            itemBeingFlaggedType = valueitemBeingFlaggedType;
        } else {
            itemBeingFlaggedType = null;
        }
        String valueitemBeingFlaggedId = request.getQueryString("itemBeingFlaggedId");
        Long itemBeingFlaggedId;
        if (valueitemBeingFlaggedId != null) {
            itemBeingFlaggedId = Long.parseLong(valueitemBeingFlaggedId);
        } else {
            itemBeingFlaggedId = null;
        }
        String valueflagableType = request.getQueryString("flagableType");
        String flagableType;
        if (valueflagableType != null) {
            flagableType = valueflagableType;
        } else {
            flagableType = null;
        }
        String valueflagableId = request.getQueryString("flagableId");
        Long flagableId;
        if (valueflagableId != null) {
            flagableId = Long.parseLong(valueflagableId);
        } else {
            flagableId = null;
        }
        return imp.deleteFlagHttp(request, version, deviceId, accountId, itemBeingFlaggedType, itemBeingFlaggedId, flagableType, flagableId);
    }

    @ApiAction
    public Result getFlag(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueflagableType = request.getQueryString("flagableType");
        String flagableType;
        if (valueflagableType != null) {
            flagableType = valueflagableType;
        } else {
            throw new IllegalArgumentException("'flagableType' parameter is required");
        }
        String valueflagableId = request.getQueryString("flagableId");
        Long flagableId;
        if (valueflagableId != null) {
            flagableId = Long.parseLong(valueflagableId);
        } else {
            throw new IllegalArgumentException("'flagableId' parameter is required");
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.getFlagHttp(request, version, flagableType, flagableId, deviceId, accountId, latitude, longitude);
    }

    @ApiAction
    public Result getFlagThreshold(Http.Request request, BigDecimal version) throws Exception {
        String valueitemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType");
        String itemBeingFlaggedType;
        if (valueitemBeingFlaggedType != null) {
            itemBeingFlaggedType = valueitemBeingFlaggedType;
        } else {
            throw new IllegalArgumentException("'itemBeingFlaggedType' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        return imp.getFlagThresholdHttp(request, version, itemBeingFlaggedType, appKey);
    }

    @ApiAction
    public Result updateFlagThreshold(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueitemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType");
        String itemBeingFlaggedType;
        if (valueitemBeingFlaggedType != null) {
            itemBeingFlaggedType = valueitemBeingFlaggedType;
        } else {
            throw new IllegalArgumentException("'itemBeingFlaggedType' parameter is required");
        }
        String valuethreshold = request.getQueryString("threshold");
        Long threshold;
        if (valuethreshold != null) {
            threshold = Long.parseLong(valuethreshold);
        } else {
            throw new IllegalArgumentException("'threshold' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        return imp.updateFlagThresholdHttp(request, version, itemBeingFlaggedType, threshold, appKey, deviceId, accountId);
    }

}

File: server-scala-play-server/app/api/FlagApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.CountResponse
import model.FlagResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class FlagApiController @Inject()(cc: ControllerComponents, api: FlagApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/flag/create?deviceId=[value]&accountId=[value]&flagableType=[value]&flagableId=[value]&flagDescription=[value]&latitude=[value]&longitude=[value]
    */
  def createFlag(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val flagableType = request.getQueryString("flagableType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("flagableType", "query string")
        }
        
      val flagableId = request.getQueryString("flagableId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("flagableId", "query string")
        }
        
      val flagDescription = request.getQueryString("flagDescription")
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.createFlag(flagableType, flagableId, deviceId, accountId, flagDescription, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/flag/delete?deviceId=[value]&accountId=[value]&itemBeingFlaggedType=[value]&itemBeingFlaggedId=[value]&flagableType=[value]&flagableId=[value]
    */
  def deleteFlag(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val itemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType")
        
      val itemBeingFlaggedId = request.getQueryString("itemBeingFlaggedId")
        .map(value => value.toLong)
        
      val flagableType = request.getQueryString("flagableType")
        
      val flagableId = request.getQueryString("flagableId")
        .map(value => value.toLong)
        
      api.deleteFlag(deviceId, accountId, itemBeingFlaggedType, itemBeingFlaggedId, flagableType, flagableId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/flag/get?deviceId=[value]&accountId=[value]&flagableType=[value]&flagableId=[value]&latitude=[value]&longitude=[value]
    */
  def getFlag(): Action[AnyContent] = Action { request =>
    def executeApi(): FlagResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val flagableType = request.getQueryString("flagableType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("flagableType", "query string")
        }
        
      val flagableId = request.getQueryString("flagableId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("flagableId", "query string")
        }
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.getFlag(flagableType, flagableId, deviceId, accountId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/flag/threshold/get?itemBeingFlaggedType=[value]&appKey=[value]
    */
  def getFlagThreshold(): Action[AnyContent] = Action { request =>
    def executeApi(): CountResponse = {
      val itemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("itemBeingFlaggedType", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      api.getFlagThreshold(itemBeingFlaggedType, appKey)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/flag/threshold/update?deviceId=[value]&accountId=[value]&itemBeingFlaggedType=[value]&threshold=[value]&appKey=[value]
    */
  def updateFlagThreshold(): Action[AnyContent] = Action { request =>
    def executeApi(): CountResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val itemBeingFlaggedType = request.getQueryString("itemBeingFlaggedType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("itemBeingFlaggedType", "query string")
        }
        
      val threshold = request.getQueryString("threshold")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("threshold", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      api.updateFlagThreshold(itemBeingFlaggedType, threshold, appKey, deviceId, accountId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}
